estuary/utils/story.py

```diff
@@ -195,9 +197,51 @@ def get_wait_times(self, results):
 
             # Find the time between when the current artifact completes and the next one starts
             wait_time = next_artifact_start_time - completion_time
-            wait_times[index] = wait_time.seconds
+            wait_times[index] = wait_time.seconds + (wait_time.days * 86400)
+
+            # The 'wait time' between a FreshmakerEvent and a ContainerKojiBuild is still a part of
+            # the processing in a FreshmakerEvent, so we do not count it towards the total wait time
+            if artifact.__label__ != 'FreshmakerEvent':
+                total_wait_time += wait_time.seconds + (wait_time.days * 86400)
 
-        return wait_times
+        return (wait_times, total_wait_time)
+
+    def get_total_processing_time(self, results):
+        """
+        Get the total time spent processing in the pipeline.
+
+        :param list results: contains inflated results from Neo4j
+        :return: the seconds of total time spent processing
+        :rtype: int
+        """
+        len_story = len(results)
+        if len_story < 2:
+            raise RuntimeError('This function can\'t be called with one or zero elements')
+        total = 0
+        timed_processes = {
+            'FreshmakerEvent': ['time_created', 'time_done'],
+            'KojiBuild': ['creation_time', 'completion_time'],
+            'ModuleKojiBuild': ['creation_time', 'completion_time'],
+            'ContainerKojiBuild': ['creation_time', 'completion_time']
+        }
+        for index in range(len_story - 1):
+            artifact = results[index]
+            if artifact.__label__ in timed_processes.keys():
+                # We do not want the processing time of the entire FreshmakerEvent, just the
+                # processing time until the displayed ContainerKojiBuild is complete
+                if artifact.__label__ == 'FreshmakerEvent' and results[index + 1]:
+                    creation_time = getattr(artifact, timed_processes['FreshmakerEvent'][0])
+                    completion_time = getattr(results[index + 1],
+                                              timed_processes['ContainerKojiBuild'][1])
```

Comment:
Could you use `results[index + 1].__label__` instead? Bonus points if you assign `results[index + 1]` to a variable  :)

Reply:
I think we decided that the process time of a `FreshmakerEvent` is the start time of the `ContainerKojiBuild` minus the start time of the `FreshmakerEvent`, not the completion time.

Reply:
If I assign results[index+1] to a variable before it is used, couldn't it potentially lead to an out of bounds error?

Reply:
With the process time, doing this includes both the processing times of the event and the build. Since we don't need them separately, the total processing time will be the same. If the problem is with readability though, will happily change it.